config/ext_jenkins-infra_privatek8s.yaml

```diff
@@ -50,3 +51,269 @@ controller:
                     description: "Kubeconfig file for privatek8s"
                     scope: SYSTEM
                     secretBytes: "${base64:${KUBECONFIG_PRIVATEK8S}}"
+      agent-settings: |
+        jenkins:
+          numExecutors: 0
+          clouds:
+            - kubernetes:
+                containerCapStr: "100"
+                # Kubernetes internal URL to stay within private networks
+                jenkinsUrl: "http://jenkins-infra.jenkins-infra.svc.cluster.local:8080"
+                maxRequestsPerHostStr: "300"
+                webSocket: true
+                name: "kubernetes"
+                namespace: "jenkins-infra-agents"
```

Comment:
Changed the namespace and removed the credentialsId, serverUrl and serverCertificate (using the SA instead)